components/dashboard/index.js

```diff
@@ -0,0 +1,109 @@
+import styled from 'styled-components'
+import Button from '@codegouvfr/react-dsfr/Button'
+import {useState} from 'react'
+import {Line, Bar} from 'react-chartjs-2'
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js'
+
+import {useDashboardData} from '@/hooks/dashboard-data'
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend
+)
+
+const defaultChartOptions = {
+  responsive: true,
+  plugins: {
+    legend: {
+      display: false,
+    },
+    title: {
+      display: true,
+      font: {
+        size: 18
+      }
+    },
+  },
+  scales: {
+    yAxes: [{
+      display: true,
+      ticks: {
+        beginAtZero: true
+      }
+    }]
+  }
+}
+
+const DashboardContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+
+    > .dashboard-header {
+      margin-top: 15px;
+      button {
+        margin: 0 5px;
+      }
+      button.active {
+        background-color: var(--active);
+      }
+    }
+
+    > .chart-wrapper {
+      margin: 10px 0;
+      display: flex;
+      justify-content: center;
+      width: 100%;
+      height: 500px;
+    }
+`
+const timeLapseButtons = [{
+  label: 'Mois',
+  value: 30,
+}, {
+  label: 'Semaine',
+  value: 7,
+}, {
+  label: 'Jour',
+  value: 0,
+}]
+
+const Dashboard = () => {
+  const [timeLapse, setTimeLapse] = useState(30)
+  const {dashboardData} = useDashboardData(timeLapse)
+
+  return (
+    <DashboardContainer>
+      <div className='dashboard-header'>
+        {timeLapseButtons.map(({label, value}) => <Button key={label} type='button' active={value === timeLapse} className={value === timeLapse ? 'active' : ''} onClick={() => setTimeLapse(value)}>{label}</Button>)}
+      </div>
```

Comment:
J'avoue ne pas être fan de tout mettre en une ligne comme ça, ça réduit vraiment la lisibilité surtout avec un écran avec une petite résolution.

Ce n'est qu'une proposition ;) Je vous laisse voir entre vous, ce que vous préférez.

```suggestion
        {timeLapseButtons.map(({label, value}) => (
          <Button
            key={label}
            type='button'
            active={value === timeLapse}
            className={value === timeLapse ? 'active' : ''}
            onClick={() => setTimeLapse(value)}
          >
            {label}
          </Button>
        ))}
```

Reply:
done